app/screens/InvitationsScreen.js

```diff
@@ -1,78 +1,76 @@
   const [selectedInvitation, setSelectedInvitation] = useState(null);
-  const [temporaryDeviceId, setTemporaryDeviceId] = useState('');
+  const [selectedPickerItem, setSelectedPickerItem] = useState(null);
+  const [pickerItems, setPickerItems] = useState([]);
+  
+  useEffect(() => {
+    setPickerItems(devices.map(device => ({ label: device.name, value: device._id.toString() }) ));
+  }, [devices.length]);
 
   const handleAccept = async () => {
-    if (!selectedInvitation || !temporaryDeviceId)
+    if (!selectedInvitation || !selectedPickerItem)
       return;
 
     // Our MongoDB Realm backend function will return an object
     // containing an error property if any errors occurred.
-    const { error } = await respondToInvitation(selectedInvitation.groupId, true, temporaryDeviceId);
+    const selectedDeviceId = selectedPickerItem.value;
+    const { error } = await respondToInvitation(selectedInvitation.groupId, true, selectedDeviceId);
     if (error)
       return Alert.alert(error.message);
 
     setSelectedInvitation(null);
-    setTemporaryDeviceId('');
   };
 
-  const handleCancelAccept = () => {
-    setSelectedInvitation(null);
+  const handleCancelAccept = () => setSelectedInvitation(null);
 
-    if (temporaryDeviceId)
-      setTemporaryDeviceId('');
-  };
-  
   return (
     <View style={styles.screen}>
       {userData && (
         <>
-        <List
-          items={userData.invitations}
-          keyExtractor={(invitation) => invitation.groupId.toString()}
-          itemTextExtractor={(invitation) => invitation.groupName}
-          itemSubTextExtractor={(invitation) => `From: ${invitation.senderEmail}`}
-          fadeOnPress={false}
-          rightActions={[
-            {
-              actionType: 'accept',
-              onPress: (invitation) => setSelectedInvitation(invitation)
-            },
-            {
-              actionType: 'decline',
-              onPress: (invitation) => respondToInvitation(invitation.groupId, false)
-            }
-          ]}
-        />
-        <ModalForm
-          visible={!!selectedInvitation}
-          title='Accept Invitation'
-          submitText='Accept'
-          onSubmit={handleAccept}
-          onCancel={handleCancelAccept}
-        >
-          <FormTextInput
-            placeholder='Device ID to join with'
-            value={temporaryDeviceId}
-            onChangeText={setTemporaryDeviceId}
-            autoCorrect={false}
-            autoCapitalize='none'
+          <List
+            items={userData.invitations}
+            keyExtractor={(invitation) => invitation.groupId.toString()}
+            itemTextExtractor={(invitation) => invitation.groupName}
+            itemSubTextExtractor={(invitation) => `From: ${invitation.senderEmail}`}
+            fadeOnPress={false}
+            rightActions={[
+              {
+                actionType: 'accept',
+                onPress: (invitation) => setSelectedInvitation(invitation)
```

Comment:
Have you considered memoizing these with `useCallback` too?
I would consider it a must-have, since it's not a hot path of the code, but then again - all performance matters if the fix is quick and not too ugly.